scripts/pre_commit_linter.py

```diff
@@ -2188,42 +2207,99 @@ def _check_bad_patterns(self):
 
         return summary_messages
 
+    def check_for_important_rules_at_bottom_of_codeowners(
+            self, important_patterns):
+        """Checks that the most important rules/patterns are at the bottom
+        of the CODEOWNERS file.
+
+        Arguments:
+            important_patterns: list(str). List of the important
+                paths/rules for CODEOWNERS file.
+
+        Returns:
+            bool. Status of failure.
+        """
+
+        failed = False
+        # Check that there are no duplicate elements in the lists.
+        important_patterns_set = set(important_patterns)
+        codeowner_important_paths_set = set(CODEOWNER_IMPORTANT_PATHS)
+        if len(important_patterns_set) != len(important_patterns):
+            print('%s --> Duplicate patterns found in critical rules'
+                  ' section.' % CODEOWNER_FILEPATH)
+            failed = True
+        if len(codeowner_important_paths_set) != len(CODEOWNER_IMPORTANT_PATHS):
+            print('scripts/pre_commit_linter.py --> Duplicate pattern found in '
```

Comment:
Nit: pattern --> patterns for consistency with previous message?

Reply:
Done!